primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.selectcheckboxmenu.js

```diff
@@ -194,31 +197,39 @@ PrimeFaces.widget.SelectCheckboxMenu = PrimeFaces.widget.BaseWidget.extend({
      * @private
      */
     renderPanel: function() {
-        this.panel = $('<div id="' + this.panelId + '" class="ui-selectcheckboxmenu-panel ui-widget ui-widget-content ui-corner-all ui-helper-hidden ui-input-overlay" role="dialog"></div>');
-        this.transition = PrimeFaces.utils.registerCSSTransition(this.panel, 'ui-connected-overlay');
-        if (this.cfg.panelStyle) {
-            this.panel.attr('style', this.cfg.panelStyle);
-        }
-        if (this.cfg.panelStyleClass) {
-            this.panel.addClass(this.cfg.panelStyleClass);
-        }
-        this.cfg.appendTo = PrimeFaces.utils.resolveAppendTo(this, this.jq, this.panel);
+        var rawPanelId = this.id + '_panel';
 
-        PrimeFaces.utils.registerDynamicOverlay(this, this.panel, this.id + '_panel');
+        //init panel content rendered by SelectCheckboxMenuRenderer
+        this.header = this.panel.children('.ui-selectcheckboxmenu-header');
+        this.toggler = this.header.children('.ui-chkbox');
+        this.togglerBox = this.toggler.children('.ui-chkbox-box');
+        this.filterInputWrapper = this.header.children('.ui-selectcheckboxmenu-filter-container');
+        this.filterInput = this.filterInputWrapper.children('.ui-inputtext');
+        this.closer = this.header.children('.ui-selectcheckboxmenu-close');
 
-        if (this.cfg.showHeader) {
-            this.renderHeader();
+        if (this.cfg.renderPanelContentOnClient && this.itemContainerWrapper.children().length === 0) {
+            this.renderItems();
+        }
+        else {
+            //init items rendered by SelectCheckboxMenuRenderer
+            this.itemContainer = this.itemContainerWrapper.children('.ui-selectcheckboxmenu-items');
+            this.items = this.itemContainer.find('.ui-selectcheckboxmenu-item');
+            this.groupHeaders = this.itemContainer.find('.ui-selectcheckboxmenu-item-group');
         }
-
-        this.renderItems();
 
         if (this.cfg.scrollHeight) {
-            this.itemContainerWrapper.height(this.cfg.scrollHeight);
+            this.itemContainerWrapper.css('max-height', this.cfg.scrollHeight);
```

Comment:
Are these height changes relevant to column support? Looks like a different issue to me.